src/freedesktop.rs

```diff
@@ -951,6 +990,42 @@ mod tests {
         }
     }
 
+    #[test]
+    fn uri_enc_dec_roundtrip() {
+        let fake = format!("/tmp/{}", get_unique_name());
+        let path = decode_uri_path(&fake);
+
+        assert_eq!(fake, path.to_str().expect("Path is valid Unicode"), "Decoded path shouldn't be different");
```

Comment:
Generally, I go with `assert_eq(actual, expected)`, but it might be there is precedent for doing it the other way around somewhere in the codebase already. But if not, let's go with `(actual, expected)` instead.